### PR TITLE
[ROCKETMQ-38] Some unit tests for rocketmq-remoting

### PR DESCRIPTION
**Pull request by iskl**
[ref issue](https://issues.apache.org/jira/browse/ROCKETMQ-38).

I wish to improve unit test coverage for RocketMQ and wrote these **sample** tests following @vongosling 's guidance. 

> 1. Using testMethod_Condition naming conversion,
Because it's more friendly for maven surefire and idea moreunit plugin
> 2. Mock the module's interaction parts. 
We can use mockito tool to do this
> 3. More assert instead of simple printting. 
Recommend to use assert4j tool
> 4. Do not throw exception in test method
> 5. Use latch instead of thead.sleep
> 6. Pay more attention for your test time-consuming

@vongosling , @zhouxinyu , Would you mind review these **sample** test code to see whether the code style is suitable? Peg for suggestion. If that's OK, The remaining part would be finished just like the sample test code.

### CODE REVIEW

**Comment by zhouxinyu** [comment]
Hi @iskl, please make sure the PR can pass the CI test.

**Comment by iskl** [comment]
@zhouxinyu Apologize... It's my mistake to use a wrong version of assertj.-core I've modified it to the correct version for jdk7 :)

File: remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+
+        try {
+            CommandCustomHeader decodedHeader = decodedCommand.decodeCommandCustomHeader(ExtFieldsHeader.class);
+            assertThat(((ExtFieldsHeader)decodedHeader).getStringValue()).isEqualTo("bilibili");
+            assertThat(((ExtFieldsHeader)decodedHeader).getIntValue()).isEqualTo(2333);
+            assertThat(((ExtFieldsHeader)decodedHeader).getLongValue()).isEqualTo(23333333l);
+            assertThat(((ExtFieldsHeader)decodedHeader).isBooleanValue()).isEqualTo(true);
+            assertThat(((ExtFieldsHeader)decodedHeader).getDoubleValue()).isBetween(0.617, 0.619);
+
+        } catch (RemotingCommandException ex) {
```
Comment (WillemJiang) [inline remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java]: We don't need to catch the exception here.  The unit test can be failed when the exception is thrown.

Comment (iskl) [inline remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java]: Thanks for your correction... I will keep in mind in my future work...

File: remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        assertThat(((ExtFieldsHeader) decodedHeader).getLongValue()).isEqualTo(23333333l);
+        assertThat(((ExtFieldsHeader) decodedHeader).isBooleanValue()).isEqualTo(true);
+        assertThat(((ExtFieldsHeader) decodedHeader).getDoubleValue()).isBetween(0.617, 0.619);
+    }
+}
+
+class SampleCommandCustomHeader implements CommandCustomHeader {
+    @Override
+    public void checkFields() throws RemotingCommandException {
+        return;
```
Comment (zhouxinyu) [inline remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java]: I think we can remove the empty return.

**Comment by zhouxinyu** [comment]
It seems ok, please @vongosling @lizhanhui help review.

**Comment by shroman** [comment]
A suggestion: how about renaming this JIRA issue? This issue is for RemotingCommand, _'some'_ is very vague.
We can end up with many more _'some'_ tests in future :)

**Comment by iskl** [comment]
@shroman , issue-38 is for the whole rocketmq-remoting submoudle. The code already committed is the sample just for code style suggestion. I am writing the remaining test cases and nearly complete. For other submodules, I will create new issues~ :)

File: remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java (hunk 2 above)
Comment (iskl) [inline remoting/src/test/java/org/apache/rocketmq/remoting/protocol/RemotingCommandTest.java]: You're right~ I will improve it in next commit~

**Comment by vongosling** [comment]
Thanks @iskl @shroman @WillemJiang , I will work on the rocketmq remoting module these days.

**Pull request closed without merging**